payments-core/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -160,8 +161,18 @@ internal class StripePaymentController internal constructor(
     ) {
         logReturnUrl(confirmStripeIntentParams.returnUrl)
 
-        val returnUrl = confirmStripeIntentParams.returnUrl.takeUnless { it.isNullOrBlank() }
-            ?: defaultReturnUrl.value
+        val returnUrl =
+            shouldOverrideReturnUrl(
+                confirmStripeIntentParams,
+                requestOptions
+            ).let { shouldOverrideReturnUrl ->
+                if (shouldOverrideReturnUrl) {
+                    confirmStripeIntentParams.returnUrl.takeUnless { it.isNullOrBlank() }
+                        ?: defaultReturnUrl.value
+                } else {
+                    confirmStripeIntentParams.returnUrl
+                }
+            }
```

Comment:
From https://stripe.com/docs/api/payment_intents/confirm#confirm_payment_intent-return_url

> This parameter is only used for cards and other redirect-based payment methods.

Do you think it makes sense to invert this logic? Should we only add `return_url` when it's a card or redirect-based payment method?